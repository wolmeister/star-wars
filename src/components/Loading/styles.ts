import styled, { keyframes } from 'styled-components';

// css by https://tobiasahlin.com/spinkit/

const bounceKeyframes = keyframes`
  0%, 100% {
    transform: scale(0.0);
    -webkit-transform: scale(0.0);
  } 50% {
    transform: scale(1.0);
    -webkit-transform: scale(1.0);
  }
`;

export const StyledLoading = styled.div`
  position: absolute;
  top: calc(50% - 20px);
  left: calc(50% - 20px);
`;

export const Spinner = styled.div`
  width: 40px;
  height: 40px;

  position: relative;
`;

export const BounceA = styled.div`
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: ${props => props.theme.colors.primary};
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;

  animation: ${bounceKeyframes} 2s infinite ease-in-out;
`;

export const BounceB = styled(BounceA)`
  animation-delay: -1s;
`;
