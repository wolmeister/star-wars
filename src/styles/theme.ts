import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  colors: {
    background: '#000000',
    primary: '#ffff00',
    text: '#ffffff',
  },
};

export default theme;
