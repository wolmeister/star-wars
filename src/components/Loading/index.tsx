import React from 'react';

import { StyledLoading, Spinner, BounceA, BounceB } from './styles';

export default function Loading() {
  return (
    <StyledLoading>
      <Spinner>
        <BounceA />
        <BounceB />
      </Spinner>
    </StyledLoading>
  );
}
