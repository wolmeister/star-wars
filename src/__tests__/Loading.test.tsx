import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import Loading from '~/components/Loading';
import theme from '~/styles/theme';

describe('Loading', () => {
  test('renders correctly', () => {
    const loading = render(
      <ThemeProvider theme={theme}>
        <Loading />
      </ThemeProvider>
    );
    expect(loading).toMatchSnapshot();
  });
});
