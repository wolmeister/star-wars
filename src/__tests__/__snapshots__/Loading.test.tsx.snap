// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: absolute;
  top: calc(50% - 20px);
  left: calc(50% - 20px);
}

.c1 {
  width: 40px;
  height: 40px;
  position: relative;
}

.c2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #ffe81f;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-animation: eogrdy 2s infinite ease-in-out;
  animation: eogrdy 2s infinite ease-in-out;
}

.c3 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #ffe81f;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-animation: eogrdy 2s infinite ease-in-out;
  animation: eogrdy 2s infinite ease-in-out;
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          />
          <div
            class="c3"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa kMUEOT"
    >
      <div
        class="sc-bwzfXH kxqXaa"
      >
        <div
          class="sc-htpNat iQoukG"
        />
        <div
          class="sc-htpNat sc-bxivhb gJFXHR"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
